### PR TITLE
Remove split if first element is empty to create anon volume

### PR DESCRIPTION
**Pull request by manugupt1**
Fixes https://github.com/containerd/nerdctl/issues/1378

Signed-off-by: Manu Gupta <manugupt1@gmail.com>

### CODE REVIEW

**Comment by Zheaoli** [comment]
Would you mind to add extra unit test?

File: pkg/mountutil/mountutil.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
 	)
 
 	split := strings.Split(s, ":")
+	if split[0] == "" {
+		split = split[1:]
+	}
```
Comment (fahedouch) [inline pkg/mountutil/mountutil.go:62]: wrong result  for `nerdctl run  --mount type=volume,dst=/foo,readonly`

**Comment by AkihiroSuda** [comment]
> Would you mind to add extra unit test?

Needs an integration test

**Comment by manugupt1** [comment]
This would require a while; I did not account for the exit code. I will fix it in a few days.

File: pkg/mountutil/mountutil.go (hunk 1 above)
Comment (manugupt1) [reply pkg/mountutil/mountutil.go]: This does not create for docker as well; however I have made the code changes for this as well.

Comment (manugupt1) [reply pkg/mountutil/mountutil.go]: Sorry for going back and forth; adding incompatible changes did not make much sense as the code there was fragile. Keeping it simple

Comment (manugupt1) [reply pkg/mountutil/mountutil.go]: FYI:
```
➜  nerdctl git:(anon-volume) ✗ docker run --mount type=volume,dst=/foo,readonly alpine
docker: Error response from daemon: invalid mount config for type "volume": must not set ReadOnly mode when using anonymous volumes.
See 'docker run --help'.
➜  nerdctl git:(anon-volume) ✗ docker run --mount type=volume,dst=/foo,rw alpine      
invalid argument "type=volume,dst=/foo,rw" for "--mount" flag: invalid field 'rw' must be a key=value pair
See 'docker run --help'.
➜  nerdctl git:(anon-volume) ✗ 
```

**Comment by manugupt1** [comment]
Ready for review.Ptal

File: pkg/mountutil/mountutil.go [hunk 2]
```diff
@@ -57,9 +57,12 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
 	)
 
 	split := strings.Split(s, ":")
-	switch len(split) {
-	case 1:
+
+	if len(split) == 1 || split[0] == "" {
```
Comment (junnplus) [inline pkg/mountutil/mountutil.go:61]: Why not just remove the leading characters ":" ?

Comment (manugupt1) [reply pkg/mountutil/mountutil.go:61]: Sure, we can do that. I just wanted to isolate under this block. Let me know if you prefer stripping

**Review by junnplus — APPROVED** [review]
Thanks, LGTM

File: pkg/mountutil/mountutil.go [hunk 3]
```diff
@@ -56,6 +56,7 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
 		options  []string
 	)
 
+	s = strings.TrimLeft(s, ":")
```
Comment (AkihiroSuda) [inline pkg/mountutil/mountutil.go:59]: I'm confused. This code seems for parsing `-v` but the added test uses `--mount`.

Comment (junnplus) [reply pkg/mountutil/mountutil.go:59]: `--mount` also deal with here
https://github.com/containerd/nerdctl/blob/7dbbc311506c70edcfea9e4f300b4e1309c3ddda/pkg/mountutil/mountutil_linux.go#L418-L427

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**